dlt/destinations/impl/duckdb/duck.py

```diff
@@ -123,28 +119,20 @@ def run(self) -> None:
 
         qualified_table_name = self._sql_client.make_qualified_table_name(self.load_table_name)
         if self._file_path.endswith("parquet"):
-            source_format = "PARQUET"
-            options = ""
-            # lock when creating a new lock
-            with PARQUET_TABLE_LOCK:
-                # create or get lock per table name
-                lock: threading.Lock = TABLES_LOCKS.setdefault(
-                    qualified_table_name, threading.Lock()
-                )
+            source_format = "read_parquet"
+            options = ", union_by_name=true"
         elif self._file_path.endswith("jsonl"):
             # NOTE: loading JSON does not work in practice on duckdb: the missing keys fail the load instead of being interpreted as NULL
```

Comment:
Judging from the work on my datasets PR this is not or no longer true. I have a test there that migrates a table and it still works in duckdb.